src/admin/api.js

```diff
@@ -349,6 +353,102 @@ function extractDestinationCurrency(currencyPair) {
     return currencyPair.substring(3, 6);
 }
 
+/**
+ * @function buildDecimalRate
+ * @param {String} rate
+ * @param {Number} decimalPlaces
+ * @returns {String}
+ */
+function buildDecimalRate(rate, decimalPlaces) {
```

Comment:
I worry that a lot of this very difficult-to-get-right functionality is currently duplicated on the finance-portal-ui. Try to see if you can extract it from there first before re-implementing here.

Reply:
I'm flagging @msk- just to relive flashbacks about how many iterations the other code paths had to go through to fix edge cases and issues.

Reply:
I checked it, the last part is the same as the implementation of this function, but before that it has some logic for adding zeroes, which is not needed here but is fitting only to the UI part.